File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 0:
            return 0
        
        stack = []
        for i in s:
            if i == "(" or i == "{" or i == "[":
                stack.append(i)
            if i ==")":
                if len(stack)>0 and stack[-1] == "(":
                    stack.pop(-1)
                else:
                    return False
            if i =="}":
                if len(stack)>0 and stack[-1] == "{":
                    stack.pop(-1)
                else:
                    return False
                
            if i =="]":
                if len(stack)>0 and stack[-1] == "[":
                    stack.pop(-1)
                else:
                    return False
            
        
        if len(stack) == 0:
            return True
        else: return False

File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        
        if (s.length() == 0){
            return false;
        }
        
        Stack <Character> stack = new Stack<>();
        
        for(int i=0; i< s.length(); i++)
        {
            char c = s.charAt(i);
            if ( c == '(' || c == '{' || c == '[')
            {
                stack.push(c);
                continue;
                
            }
            
            if (c == ')'){
                if( !stack.empty() && stack.peek() == '(' ){
                    stack.pop();
                }
                else{
                    return false;
                }
            }
            
            if (c == ']'){
                if( !stack.empty() && stack.peek() == '[' ){
                    stack.pop();
                }
                else{
                    return false;
                }
            }
            
            if (c == '}'){
                if( !stack.empty() && stack.peek() == '{'){
                    stack.pop();
                }
                else{
                    return false;
                }
            }
            
            
            
            
        }
        
        
        if (stack.empty()){
            return true;
        }
        else{
            return false;
        }
        
        
    }
}